lib/src/test/groovy/com/cloudbees/groovy/cps/impl/FunctionCallBlockTest.groovy

```diff
@@ -21,6 +22,28 @@ class FunctionCallBlockTest extends AbstractGroovyCpsTest {
             throw new NamingException();
     }
 
+    /** Need to run more steps so we can hit the recursion limit */
+    Object evalCPSonly(String script, int numSteps) {
+        return parseCps(script).invoke(null, null, Continuation.HALT).run(numSteps).replay()
+    }
+
+    @Test
+    void infiniteRecursion() {
+        try {
+            assert evalCPSonly("""
+                def thing = null;
+                def getThing() { 
+                    return thing == null;
+                }
+                def stuff = getThing();
+            """, 25000) == "cheese";
+        } catch (StackOverflowError soe) {
```

Comment:
You need to `fail` if the error is _not_ thrown.

Reply:
Fails implicitly, the above never returns "cheese" ;)